python/cugraph/cugraph/gnn/data_loading/bulk_sampler.py

```diff
@@ -196,31 +216,62 @@ def flush(self) -> None:
             self.__sample_call_args.update(
                 {
                     "_multiple_clients": True,
-                    "label_to_output_comm_rank": self.__get_label_to_output_comm_rank(
-                        min_batch_id, max_batch_id
-                    ),
-                    "label_list": cupy.arange(
-                        min_batch_id, max_batch_id + 1, dtype="int32"
-                    ),
+                    "keep_batches_together": True,
+                    "min_batch_id": min_batch_id,
+                    "max_batch_id": max_batch_id,
                 }
             )
 
+        start_time_sample_call = time.perf_counter()
+
+        # Call uniform neighbor sample
         samples, offsets = sample_fn(
             self.__graph,
             **self.__sample_call_args,
-            start_list=self.__batches[self.start_col_name][batch_id_filter],
-            batch_id_list=self.__batches[self.batch_col_name][batch_id_filter],
+            start_list=self.__batches[[self.start_col_name, self.batch_col_name]][
+                batch_id_filter
+            ],
+            with_batch_ids=True,
             with_edge_properties=True,
             return_offsets=True,
         )
 
+        end_time_sample_call = time.perf_counter()
+        sample_runtime = end_time_sample_call - start_time_sample_call
+
+        self.__logger.info(
+            f"Called uniform neighbor sample, took {sample_runtime:.4f} s"
+        )
+
+        start_time_filter_batches = time.perf_counter()
+
+        # Filter batches to remove those already processed
         self.__batches = self.__batches[~batch_id_filter]
-        if isinstance(self.__batches, dask_cudf.DataFrame):
+        del batch_id_filter
+        if hasattr(self.__batches, "compute"):
             self.__batches = self.__batches.persist()
 
+        end_time_filter_batches = time.perf_counter()
+        self.__logger.info(
+            "Filtered batches, took "
+            f"{end_time_filter_batches - start_time_filter_batches} s"
+        )
```

Comment:
This time is not accurate as we are not waiting here as persist is not a sync operation. I would probably just remove this.